feat(native): screenshot capture (windows) (#13064)
getsentry/sentry-docs: docs/platforms/native/common/enriching-events/screenshots/index.mdx

---
title: "Screenshots"
description: "Learn more about taking screenshots when an error occurs. Sentry pairs the screenshot with the original event, giving you additional insight into issues."
---

> Sentry makes it possible to automatically take a screenshot and include it as an <PlatformLink to="/enriching-events/attachments/">attachment</PlatformLink> when a user experiences an error, an exception or a crash.

Reviewer: ```suggestion
Sentry makes it possible to automatically take a screenshot and include it as an <PlatformLink to="/enriching-events/attachments/">attachment</PlatformLink> when a user experiences an error, an exception, or a crash.
```

> This feature is only available on Windows. It's also limited by whether taking a screenshot is possible or not. For example, the `inproc` and `breakpad` backends use local exception handlers for capturing the screenshot and therefore rely on the exception handler being called. Only the `crashpad` backend with an external handler process is able to capture screenshots of fast-fail crashes that bypass Structured Exception Handling ([`SEH`](https://learn.microsoft.com/en-us/windows/win32/debug/structured-exception-handling)). Another example where a screenshot might not be available is when the event happens before the screen starts to load. So inherently, this feature is a best effort solution.

Reviewer: ```suggestion
This feature is currently only available on Windows and has a few limitations. For example, the `inproc` and `breakpad` backends use local exception handlers for capturing the screenshot and therefore rely on the exception handler being called. Only the `crashpad` backend with an external handler process is able to capture screenshots of fast-fail crashes that bypass Structured Exception Handling ([`SEH`](https://learn.microsoft.com/en-us/windows/win32/debug/structured-exception-handling)). Another example where a screenshot might not be available is when the event happens before the screen starts to load. So inherently, this feature is a best effort solution.
```

---
title: "Screenshots"
description: "Learn more about taking screenshots when an error occurs. Sentry pairs the screenshot with the original event, giving you additional insight into issues."
---

Sentry makes it possible to automatically take a screenshot and include it as an <PlatformLink to="/enriching-events/attachments/">attachment</PlatformLink> when a user experiences an error, an exception or a crash.

> This feature is only available for SDKs with a user interface, like the ones for mobile and desktop applications. It's also limited by whether taking a screenshot is possible or not. For example, in some environments, like native iOS, taking a screenshot requires the UI thread, which often isn't available in the event of a crash. Another example where a screenshot might not be available is when the event happens before the screen starts to load. So inherently, this feature is a best effort solution.

Reviewer 2: Text like this would typically end up in a non-platform-specific template, so I am surprised to see it here. Did you take this from an SDK that depends on multiple platform upstream SDKs? Then, I would understand why the text is formulated like this (since the SDK would defer screenshotting to upstream and whether it works depends on the platform it's running on).

CC: @kahest

Author: There are several copies of the same page with some minor tweaks. Not sure if there have been plans to unify this but perhaps it would be better to start with that...

- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/android/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/apple/common/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/dart/guides/flutter/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/dotnet/guides/maui/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/dotnet/guides/xamarin/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/javascript/guides/electron/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/kotlin/guides/kotlin-multiplatform/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/react-native/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/unity/enriching-events/screenshots/index.mdx
- https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/unreal/enriching-events/screenshots/index.mdx

Reviewer 3: for context: originally, this was one (template) page that was inherited by all these platforms. at some point, we broke up this inheritance in favor of duplication (which had a bunch of benefits, but the screenshot docs were not one of them). that's why we have duplicated docs.

that said, IMO it's ok for this PR to duplicate the content tweak it as necessary to stay correct/make sense for Native (like we did for the other pages).

we have a separate [product page for Screenshots](https://docs.sentry.io/product/issues/issue-details/#screenshot) - we should add Native to the list there. at some point it'd also make sense to extract the commonalities/generic aspects from the platform docs to this page, but let's not do it in this PR

Reviewer 3: re. tweaking the content for Native - if there's specific limitations of when/how screenshots may or may not work in this specific SDK/impl, this is a good place to call it out. we can replace the current callouts to UI thread etc. with whatever makes sense. e.g. I think the note about it only working on Windows would fit here

Author: > we have a separate [product page for Screenshots](https://docs.sentry.io/product/issues/issue-details/#screenshot) - we should add Native to the list there.

Oops sorry, almost forgot this! I've added the missing link to Native.
